packages/modular-scripts/src/build.ts

```diff
@@ -197,15 +198,33 @@ async function makeBundle(
       `${packagesRoot}/${packagePath} is marked private, bailing...`,
     );
   }
-  if (!packageJson.main) {
-    throw new Error(
-      `package.json at ${packagesRoot}/${packagePath} does not have a "main" field, bailing...`,
-    );
+
+  let compilingBin = false;
+  let main: string | undefined;
+
+  if (packageJson.main) {
+    main = packageJson.main;
+  } else {
+    if (packageJson.bin) {
+      const bins: string[] = Object.values(packageJson.bin) as string[];
+      if (bins.length === 1) {
+        compilingBin = true;
```

Comment:
dropping this todo for me to handle later 

```suggestion
        // TODO: hoist the logic that determines `compilingBin` outside of makeBundle
        compilingBin = true;
```